src/components/App.js

```diff
@@ -1,89 +1,57 @@
-import React, { Component } from 'react';
-import { connect } from 'react-redux';
+import React, { useEffect } from 'react';
+import qs from 'qs';
 import { withTranslation } from 'react-i18next';
+import { Context, PermissionLevel } from '@graasp/sdk';
+import { useLocalContext } from '@graasp/apps-query-client';
 import PropTypes from 'prop-types';
 import StudentMode from './modes/student/StudentMode';
-import { getContext } from '../actions';
-import { DEFAULT_LANG, DEFAULT_MODE, MODES } from '../config/settings';
-import { DEFAULT_VIEW } from '../config/views';
+import { DEFAULT_LANG } from '../config/settings';
 import TeacherMode from './modes/teacher/TeacherMode';
-import ProgressScreen from './common/LoadingScreen';
 
-export class App extends Component {
-  static propTypes = {
-    i18n: PropTypes.shape({
-      defaultNS: PropTypes.string,
-      changeLanguage: PropTypes.func,
-    }).isRequired,
-    lang: PropTypes.string,
-    mode: PropTypes.string,
-    view: PropTypes.string,
-    loading: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    lang: DEFAULT_LANG,
-    mode: DEFAULT_MODE,
-    view: DEFAULT_VIEW,
-    loading: true,
-  };
-
-  componentDidMount() {
-    const { lang } = this.props;
-    // set the language on first load
-    this.handleChangeLang(lang);
-  }
-
-  componentDidUpdate({ lang: prevLang }) {
-    const { lang } = this.props;
-    // handle a change of language
-    if (lang !== prevLang) {
-      this.handleChangeLang(lang);
-    }
-  }
+export const App = ({ i18n }) => {
```

Comment:
You can get the current i18n instance with the `useTransations` hook if that can be useful ?